dna/keyRegistration/keyRegistration.js

```diff
@@ -1,35 +1,92 @@
 function genesis(){
-  keyRegistrationCreate("1");
+  keyRegistrationCreate();
   return true;
 }
 
-function keyRegistrationCreate(choice){
+function keyRegistrationCreate(){
+  debug("select Revocation Methord");//TODO for now the default selection will be "1" i.e the revocation_key Methord
   debug("Creating the user's keyRegistration");
-  //debug("choice :"+choice);
-  keyRegistration={perm_dpki_id:App.Agent.Hash,public_key:App.Key.Hash,shared_ID:App.Agent.String,revocation_Method_ID:choice};
-  //debug("keyRegistration="+keyRegistration);
-  directory=getDirectory();
+  revocation_Method_ID=selectRevocationMethord();
+  keyRegistration={perm_dpki_id:App.Agent.TopHash,public_key:App.Key.Hash,shared_ID:App.Agent.String,revocation_Method_ID:revocation_Method_ID};
```

Comment:
I think this is backwards in that the way I did it is that the permanent agent ID is App.Agent.Hash (which is always the second entry in the chain).  App.Agent.TopHash is the most recent agent entry committed by a call to updateAgent.  Use  Agent.TopHash for the most up-to-date Identity value, and Agent.Hash for the linking location.